packages/common/src/authentication/AuthContext.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { createContext, FC, useEffect, useMemo, useRef } from 'react';
+import { Store, User } from '@common/types';
+import { useLocalStorage } from '../localStorage';
+import { useHostContext } from '../hooks/useHostContext';
+import Cookies from 'js-cookie';
+import { addMinutes, differenceInMinutes } from 'date-fns';
+import { useOmSupplyApi } from '../api';
+import { useGetRefreshToken } from './hooks';
+
+const COOKIE_LIFETIME_MINUTES = 60;
+
+interface AuthCookie {
+  expires?: Date;
+  store?: Store;
+  token: string;
+  user?: User;
+}
+
+interface AuthControl {
+  onLoggedIn: (user: User, token: string, store?: Store) => void;
+  storeId?: string;
+  token?: string;
+  user?: User;
+  store?: Store;
+}
+
+export const getAuthCookie = (): AuthCookie => {
+  const authString = Cookies.get('auth');
+  const emptyCookie = { token: '' };
+  if (!!authString) {
+    try {
+      const parsed = JSON.parse(authString) as AuthCookie;
+      return parsed;
+    } catch {
+      return emptyCookie;
+    }
+  }
+  return emptyCookie;
+};
+
+const useRefreshingAuth = () => {
+  const { mutate: refreshToken } = useGetRefreshToken();
+  const timeoutRef = useRef<NodeJS.Timeout>();
+
+  const { expires, token } = getAuthCookie();
+  const cookieLifetimeInMinutes = differenceInMinutes(
+    new Date(expires ?? ''),
+    new Date()
+  );
+  const timeoutInterval =
+    (cookieLifetimeInMinutes > 0 ? cookieLifetimeInMinutes : 0.3) * 60 * 1000;
+
+  React.useEffect(() => {
+    if (token) {
```

Comment:
I thought it would reschedule because the token changes - can't say for certain because I still don't have fetch sending the cookie!!
It is a very neat idea! My concern is that we refresh immediately on login and that's not very neat. Only a minor concern, I'll change anyway..